testng-core/src/main/java/org/testng/TestNG.java

```diff
@@ -843,10 +841,9 @@ public void setVerbose(int verbose) {
     m_verbose = verbose;
   }
 
-  /** This method stands deprecated as of TestNG <code>v7.9.0</code>. */
-  @Deprecated
-  public void setExecutorFactoryClass(String clazzName) {
-    this.m_executorFactory = createExecutorFactoryInstanceUsing(clazzName);
+  public void setExecutorServiceFactory(IExecutorServiceFactory factory) {
+    Objects.requireNonNull(factory);
+    m_configuration.setExecutorServiceFactory(factory);
```

Comment:
@coderabbitai  use directly the result of Objects.requireNonNull(factory) in the setter.

Reply:
@juherr - I didnt quite understand the interaction here. Are you asking @coderabbitai to automatically update this PR with what you are asking for a change or are you asking me to do it. I would rather have comments fixed by the PR creator than a bot, because it takes away the entire chain of thought when a bot fiddles around with a PR. Please help me understand as to what is the intent here.